azurelinuxagent/common/cgroupapi.py

```diff
@@ -241,17 +246,108 @@ class SystemdCgroupsApi(CGroupsApi):
     """
     Cgroups interface via systemd
     """
+
+    @staticmethod
+    def create_and_start_unit(unit_filename, unit_contents):
+        try:
+            unit_path = os.path.join(UNIT_FILES_FILE_SYSTEM_PATH, unit_filename)
+            fileutil.write_file(unit_path, unit_contents)
+            shellutil.run_get_output("systemctl daemon-reload")
+            shellutil.run_get_output("systemctl start {0}".format(unit_filename))
+        except Exception as e:
+            raise CGroupsException("Failed to create and start {0}. Error: {1}".format(unit_filename, ustr(e)))
+
+    @staticmethod
+    def _get_extensions_slice_root_name():
+        return "system-{0}.slice".format(EXTENSIONS_ROOT_CGROUP_NAME)
+
+    @staticmethod
+    def _get_extension_slice_name(extension_name):
+        return "system-{0}-{1}.slice".format(EXTENSIONS_ROOT_CGROUP_NAME, extension_name.replace('-', '_'))
```

Comment:
you could use `_get_extension_scope_name` here to make sure that the delimiter is consistent.

```suggestion
        return "system-{0}-{1}.slice".format(EXTENSIONS_ROOT_CGROUP_NAME, self._get_extension_scope_name(extension_name))
```

Reply:
Also, let's move _get_extension_scope_name to the base class and use it in FileSystemCgroupsApi. _get_extension_cgroup_path